packages/graph-retriever/pyproject.toml

```diff
@@ -42,6 +42,7 @@ dependencies = [
     "numpy>=1.26.4",
     "typing-extensions>=4.12.2",
     "pytest>=8.3.4",
+    "immutabledict>=4.2.1",
```

Comment:
Need an immutable dict since (a) we want to freeze the dict and (b) we want to support hash (so we can put the edges in a set to dedup, etc.).